packages-common/pluggable-widgets-tools/tests/commands.js

```diff
@@ -0,0 +1,174 @@
+const { Mutex } = require("async-mutex");
+const { exec } = require("child_process");
+const { copy, readJson, writeJson } = require("fs-extra");
+const { join } = require("path");
+const { mkdir, rm, tempdir } = require("shelljs");
+const kill = require("tree-kill");
+const { promisify } = require("util");
+const { run: runYeoman } = require("yeoman-test");
+
+const LIMIT_TESTS = process.argv.includes("--limited");
+
+const CONFIGS = [
+    ["web", "ts", "8.0"],
+    ["native", "js", "8.1"],
+    ["web", "js", "8.3"],
+    ["native", "ts", "8.6"],
+    ["web", "ts", "8.6"],
+    ["web", "js", "8.7"],
+    ["web", "ts", "latest"],
+    ["native", "js", "latest"]
+];
+
+if (LIMIT_TESTS) {
+    CONFIGS.splice(1, CONFIGS.length - 2); // Remove all configs except the first and the last
+}
+
+const mutex = new Mutex();
+
+main().catch(e => {
+    console.error(e);
+    process.exit(1);
+});
+
+async function main() {
+    console.log("Preparing...");
+
+    const { stdout: packOutput } = await execAsync("npm pack --loglevel=error", join(__dirname, ".."));
+    const toolsPackagePath = join(__dirname, "..", packOutput.trim());
+
+    const results = await Promise.all(
+        CONFIGS.map(config =>
```

Comment:
It would be nice to use the async/await code style:
```
CONFIGS.map((config) => async () => {
  try {
    await runTest(...config);
    return ["success"];
  } catch (e) {
    return ["failed", e];
  }
});
```

Reply:
👍 actually can simplify it more

Reply:
Yea, indeed. In my code above you would have to invoke the function, which is an unnecessary extra step.